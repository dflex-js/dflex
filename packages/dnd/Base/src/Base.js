import { DRAGGED_ELM } from "@dflex/draggable/constants.json";
import AbstractDraggable from "@dflex/draggable/src/AbstractDraggable";

import store from "../../Store";

import { ACTIVE_PARENT } from "../../constants.json";

/**
 * Base element.
 *
 * Creates draggedElm and activeParent and initializes thresholds.
 *
 *
 * @class Base
 */
class Base extends AbstractDraggable {
  /**
   * Creates an instance of Base.
   *
   * @param {string} elementId
   * @memberof Base
   */
  constructor(elementId, clickCoordinates) {
    const {
      element,
      parent,
      branches: { siblings, parents },
    } = store.getElmTreeById(elementId);

    super(element, clickCoordinates);

    const { order } = element;

    /**
     * Initialize temp index that refers to element new position after
     * transformation happened.
     */
    this.draggedTempIndex = order.self;

    this.parentsList = parents;
    this.siblingsList = siblings;

    /**
     * Thresholds store, contains max value for each parent and for dragged. Depending on
     * ids as keys.
     */
    this.thresholds = {
      parents: {},
      dragged: {},
    };

    this.checkThresholds(this[DRAGGED_ELM]);

    /**
     * Init max direction for position
     */
    this.setThreshold(this[DRAGGED_ELM], false);

    this.setIsSingleton();

    this.setIsOrphan(parent);
  }

  /**
   * Check if dragged has no siblings and then set the related flag.
   *
   * @memberof Base
   */
  setIsSingleton() {
    /**
     * Dragged has no siblings.
     */
    this.isSingleton = !Array.isArray(this.siblingsList);
  }

  /**
   * Check if dragged has no parent and then set the related operations
   * accordingly.
   *
   * @memberof Base
   */
  setIsOrphan(parent) {
    /**
     * Dragged has no parent.
     */
    this.isOrphan = true;

    /**
     * Not all elements have parents.
     */
    if (parent) {
      /**
       * Indicator to parents that have changed. This facilitates looping in
       * affected parents only.
       */
      this.setOfTransformedIds = new Set([]);
      this.assignActiveParent(parent);
      this.setThreshold(parent, true);
      this.isOrphan = false;
    }
  }

  /**
   * Sets thresholds for dragged element position depending on its
   * position inside parent which is related to droppable left and top.
   *
   * maxDirection = current position + droppable-allowed spaces
   *
   * @memberof Base
   */
  setThreshold(droppable, isParent) {
    const { parents, dragged } = this.thresholds;

    const {
      thresholdOffset: { vertical, horizontal },
    } = this[DRAGGED_ELM];

    const {
      currentLeft,
      currentTop,
      offset: { width, height },
      id,
    } = droppable;

    const $ = isParent ? (parents[id] = {}) : dragged;

    /**
     * Overview:
     *
     * this.maxDirection: contains max value allowed for dragged to move. If dragged passes
     * these values, that means it's out.
     *
     * This threshold is related to droppable offset. If droppable changes, then
     * max/threshold changes.
     *
     * parentThreshold: to detect if dragged is moved inside or outside
     * parent. But what if we have multiple lists?
     *
     * Updating positionMax when dragged is located in new position is easy but
     * in lists we need to know all listMax(s) before entering to parent. Why?
     * because in this way we know where dragged is to detect the activeParent.
     *
     * The solution is to create parentThreshold. An object stores positionMax
     * depending on currentIndex as keys.
     */

    /**
     * Calculate max-vertical for up and down:
     */

    /**
     * When going up, currentTop decreases (-vertical).
     */
    $.maxTop = currentTop - vertical;

    /**
     * When going down, currentTop increases (+vertical) with droppable
     * taking into considerations (+ vertical).
     */
    $.maxBottom = currentTop + vertical;

    /**
     * When going left, currentLeft decreases (-horizontal).
     */
    $.maxLeft = currentLeft - horizontal;

    /**
     * When going right, currentLeft increases (+horizontal) with droppable
     * taking into considerations (+ horizontal).
     */
    $.maxRight = currentLeft + horizontal;
  }

  /**
   * Check thresholds availability and assign it to thresholds instance in
   * dragging process.
   *
   * @memberof Base
   */
  checkThresholds() {
    /**
     * Not initiating thresholdOffset for all coreInstances. Only ones which
     * dragged will be initiated.
     */
    if (!this[DRAGGED_ELM].thresholdOffset) {
      const {
        offset: { width, height },
      } = this[DRAGGED_ELM];

      /**
       * Calculates thresholdOffset only for dragged element.
       *
       * Two-thirds of the dragged element's space for vertical and horizontal. If
       * two-thirds of the dragged is out, then trigger isOut whether it is out
       * position or out parent.
       */
      this[DRAGGED_ELM].thresholdOffset = {
        vertical: Math.ceil((2 / 3) * height),
        horizontal: Math.ceil((2 / 3) * width),
      };
    }
  }

  /**
   * Add parent id to setOfTransformedIds. The goal here, is to avoid any
   * unnecessary process the parent goes through it when it's flagged as
   * transformed.
   *
   * @memberof Base
   */
  addParentAsTransformed() {
    const { id } = this[ACTIVE_PARENT];

    /**
     * Avoid adding same parents more than once using sets.
     */
    this.setOfTransformedIds.add(id);
  }

  /**
   * Assigns new ACTIVE_PARENT: parent who contains dragged
   *
   * @param {CoreInstance} coreInstance
   * @memberof Base
   */
  assignActiveParent(coreInstance) {
    /**
     * Assign instance ACTIVE_PARENT which represents droppable. Then
     * assign owner parent so we have from/to.
     */
    this[ACTIVE_PARENT] = coreInstance;

    /**
     * Add flag for undo method so we can check which  parent is being
     * transformed and which is not.
     */
    this[ACTIVE_PARENT].isTransformed = true;
    this.isDraggedOutActiveParent = false;

    /**
     * In initiating we get siblingsList. But, when dragged is in another
     * ACTIVE_PARENT we have different siblingsList now. siblingsList is
     * children of new parent. Anyway, check if there no list, get us one.
     */
    if (!this.siblingsList) {
      const {
        keys: { chK },
      } = this[ACTIVE_PARENT];

      this.siblingsList = store.getElmById(chK);
    }
  }
}

export default Base;
