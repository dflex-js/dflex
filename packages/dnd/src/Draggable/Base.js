import { DRAGGED_ELM } from "@dflex/draggable/constants.json";
import AbstractDraggable from "@dflex/draggable/src/AbstractDraggable";
import store from "../DnDStore";

import { ACTIVE_PARENT } from "../../constants.json";

/**
 * Base element.
 *
 * Creates draggedElm and activeParent and initializes thresholds.
 *
 *
 * @class Base
 */
class Base extends AbstractDraggable {
  /**
   * Creates an instance of Base.
   *
   * @param {Object} elementInstance
   * @memberof Base
   */
  constructor(elementInstance, clickCoordinates) {
    const {
      element,
      parent,
      branches: { siblings, parents },
    } = elementInstance;

    super(element, clickCoordinates);

    const { order } = element;

    /**
     * Initialize temp index that refers to element new position after
     * transformation happened.
     */
    this.tempIndex = order.self;

    this.parentsList = parents;

    /**
     * Thresholds store, contains max value for each parent and for dragged. Depending on
     * ids as keys.
     */
    this.thresholds = {
      parents: {},
      dragged: {},
    };

    this.checkThresholds(this[DRAGGED_ELM]);

    /**
     * Init max direction for position
     */
    this.setThreshold(this[DRAGGED_ELM], false);

    this.setIsSingleton(siblings);

    this.setIsOrphan(parent);

    this.dragID = store.tracker.newTravel();
  }

  /**
   * Check if dragged has no siblings and then set the related flag.
   *
   * @memberof Base
   */
  setIsSingleton(siblings) {
    if (Array.isArray(siblings)) {
      this.isSingleton = false;
      this.siblingsList = siblings;
    } else {
      this.isSingleton = true;
    }
  }

  /**
   * Check if dragged has no parent and then set the related operations
   * accordingly.
   *
   * @memberof Base
   */
  setIsOrphan(parent) {
    /**
     * Not all elements have parents.
     */
    if (parent) {
      /**
       * Indicator to parents that have changed. This facilitates looping in
       * affected parents only.
       */
      this.setOfTransformedIds = new Set([]);
      this.assignActiveParent(parent);
      this.setThreshold(parent, true);

      this.isOrphan = false;
      this.isOutActiveParent = false;
    } else {
      /**
       * Dragged has no parent.
       */
      this.isOrphan = true;
    }
  }

  /**
   * Sets thresholds for dragged element position depending on its
   * position inside parent which is related to droppable left and top.
   *
   * maxDirection = current position + droppable-allowed spaces
   *
   * @memberof Base
   */
  setThreshold(droppable, isParent) {
    // TODO: is this necessary to assign parent?

    const {
      thresholdOffset: {
        vertical: { twoThirds, third },
        horizontal,
      },
    } = this[DRAGGED_ELM];

    const { currentLeft, currentTop, id } = droppable;

    let $;

    if (isParent) {
      const {
        offset: { height },
      } = droppable;

      if (!this.thresholds.parents[id]) {
        this.thresholds.parents[id] = {};
      }

      $ = this.thresholds.parents[id];

      $.maxBottom = currentTop + height - third;
    } else {
      $ = this.thresholds.dragged;

      /**
       * When going down, currentTop increases (+vertical) with droppable
       * taking into considerations (+ vertical).
       */
      $.maxBottom = currentTop + twoThirds;
    }

    /**
     * Overview:
     *
     * this.maxDirection: contains max value allowed for dragged to move. If dragged passes
     * these values, that means it's out.
     *
     * This threshold is related to droppable offset. If droppable changes, then
     * max/threshold changes.
     *
     * parentThreshold: to detect if dragged is moved inside or outside
     * parent. But what if we have multiple lists?
     *
     * Updating positionMax when dragged is located in new position is easy but
     * in lists we need to know all listMax(s) before entering to parent. Why?
     * because in this way we know where dragged is to detect the activeParent.
     *
     * The solution is to create parentThreshold. An object stores positionMax
     * depending on currentIndex as keys.
     */

    /**
     * Calculate max-vertical for up and down:
     */

    /**
     * When going up, currentTop decreases (-vertical).
     */
    $.maxTop = currentTop - twoThirds;

    /**
     * When going left, currentLeft decreases (-horizontal).
     */
    $.maxLeft = currentLeft - horizontal;

    /**
     * When going right, currentLeft increases (+horizontal) with droppable
     * taking into considerations (+ horizontal).
     */
    $.maxRight = currentLeft + horizontal;
  }

  /**
   * Check thresholds availability and assign it to thresholds instance in
   * dragging process.
   *
   * @memberof Base
   */
  checkThresholds() {
    /**
     * Not initiating thresholdOffset for all coreInstances. Only ones which
     * dragged will be initiated.
     */
    if (!this[DRAGGED_ELM].thresholdOffset) {
      const {
        offset: { width, height },
      } = this[DRAGGED_ELM];

      /**
       * Calculates thresholdOffset only for dragged element.
       *
       * Two-thirds of the dragged element's space for vertical and horizontal. If
       * two-thirds of the dragged is out, then trigger isOut whether it is out
       * position or out parent.
       */
      this[DRAGGED_ELM].thresholdOffset = {
        vertical: {
          twoThirds: Math.ceil((2 / 3) * height),
          third: Math.ceil((1 / 3) * height),
        },
        horizontal: Math.ceil((2 / 3) * width),
      };
    }
  }

  /**
   * Add parent id to setOfTransformedIds. The goal here, is to avoid any
   * unnecessary process the parent goes through it when it's flagged as
   * transformed.
   *
   * @memberof Base
   */
  addParentAsTransformed() {
    const { id } = this[ACTIVE_PARENT];

    /**
     * Avoid adding same parents more than once using sets.
     */
    this.setOfTransformedIds.add(id);
  }

  /**
   * Assigns new ACTIVE_PARENT: parent who contains dragged
   *
   * @param {CoreInstance} coreInstance
   * @memberof Base
   */
  assignActiveParent(coreInstance) {
    /**
     * Assign instance ACTIVE_PARENT which represents droppable. Then
     * assign owner parent so we have from/to.
     */
    this[ACTIVE_PARENT] = coreInstance;

    /**
     * Add flag for undo method so we can check which  parent is being
     * transformed and which is not.
     */
    this.isOutActiveParent = false;

    /**
     * TODO: NEED TO BE UPDATED.
     *
     * In initiating we get siblingsList. But, when dragged is in another
     * ACTIVE_PARENT we have different siblingsList now. siblingsList is
     * children of new parent. Anyway, check if there no list, get us one.
     */
    // if (false) {
    //   const {
    //     keys: { chK },
    //   } = this[ACTIVE_PARENT];

    //   this.siblingsList = store.getElmById(chK);
    // }
  }
}

export default Base;
