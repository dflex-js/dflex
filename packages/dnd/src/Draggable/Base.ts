/**
 * Copyright (c) Jalal Maskoun.
 *
 * This source code is licensed under the AGPL3.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
import { AbstractDraggable } from "@dflex/draggable";

import { CoreInstanceInterface } from "@dflex/core-instance";

import type { MouseCoordinates } from "@dflex/draggable";
import type { ELmBranch } from "@dflex/dom-gen";

import store from "../DnDStore";

import type { ElmTree } from "../DnDStore";

import type {
  DraggableBaseInterface,
  ThresholdPercentages,
  LayoutThresholds,
} from "./types";
import { FinalDndOpts } from "../types";

/**
 * Base element.
 *
 * Creates draggedElm and activeParent and initializes thresholds.
 */
class Base
  extends AbstractDraggable<CoreInstanceInterface>
  implements DraggableBaseInterface
{
  tempIndex: number | null;

  operationID: string;

  opts: FinalDndOpts;

  parentsList: ELmBranch;

  siblingsList: string[] | null;

  activeParent!: CoreInstanceInterface | null;

  isOutActiveParent!: boolean;

  setOfTransformedIds!: Set<string>;

  thresholds: LayoutThresholds;

  thresholdsPercentages: ThresholdPercentages;

  constructor(
    elmTree: ElmTree,
    initCoordinates: MouseCoordinates,
    opts: FinalDndOpts
  ) {
    const {
      element,
      parent,
      branches: { siblings, parents },
    } = elmTree;

    super(element, initCoordinates);

    const { order } = element;

    this.opts = opts;

    /**
     * Initialize temp index that refers to element new position after
     * transformation happened.
     */
    this.tempIndex = order.self;

    this.parentsList = parents;

    /**
     * Thresholds store, contains max value for each parent and for dragged. Depending on
     * ids as keys.
     */
    this.thresholds = {
      siblings: {},
      dragged: {
        maxBottom: 0,
        maxTop: 0,
        maxLeft: 0,
        maxRight: 0,
      },
    };

    this.thresholdsPercentages = {
      vertical: Math.round(
        (this.opts.thresholds.vertical * this.draggedElm.offset.height) / 100
      ),
      horizontal: Math.round(
        (this.opts.thresholds.horizontal * this.draggedElm.offset.width) / 100
      ),
    };

    /**
     * Init max direction for position
     */
    this.setThreshold(this.draggedElm.currentTop, this.draggedElm.currentLeft);

    const siblingsBoundaries =
      store.siblingsBoundaries[store.registry[this.draggedElm.id].keys.sK];

    this.setThreshold(
      siblingsBoundaries.top,
      siblingsBoundaries.maxLeft,
      siblingsBoundaries.bottom,
      store.registry[this.draggedElm.id].keys.sK
    );

    this.siblingsList = Array.isArray(siblings) ? siblings : null;

    this.setIsOrphan(parent);

    this.operationID = store.tracker.newTravel();
  }

  /**
   * Check if dragged has no parent and then set the related operations
   * accordingly.
   *
   * @param parent -
   */
  private setIsOrphan(parent: CoreInstanceInterface | null) {
    /**
     * Not all elements have parents.
     */
    if (parent) {
      /**
       * Indicator to parents that have changed. This facilitates looping in
       * affected parents only.
       */
      this.setOfTransformedIds = new Set([]);
      this.assignActiveParent(parent);

      this.isOutActiveParent = false;
    } else {
      /**
       * Dragged has no parent.
       */
      this.activeParent = null;
    }
  }

  /**
   * Sets thresholds for dragged element position depending on its
   * position inside parent which is related to droppable left and top.
   *
   * @param top -
   * @param left -
   * @param height -
   * @param siblingsK -
   */
  setThreshold(top: number, left: number, height?: number, siblingsK?: string) {
    const { vertical, horizontal } = this.thresholdsPercentages;

    let $;

    if (siblingsK && height) {
      if (!this.thresholds.siblings[siblingsK]) {
        this.thresholds.siblings[siblingsK] = {
          maxBottom: 0,
          maxTop: 0,
          maxLeft: 0,
          maxRight: 0,
        };
      }

      $ = this.thresholds.siblings[siblingsK];

      $.maxBottom = height - vertical;
    } else {
      $ = this.thresholds.dragged;

      /**
       * When going down, currentTop increases (+vertical) with droppable
       * taking into considerations (+ vertical).
       */
      $.maxBottom = top + vertical;
    }

    /**
     * Calculate max-vertical for up and down:
     */

    /**
     * When going up, currentTop decreases (-vertical).
     */
    $.maxTop = top - vertical;

    /**
     * When going left, currentLeft decreases (-horizontal).
     */
    $.maxLeft = left - horizontal;

    /**
     * When going right, currentLeft increases (+horizontal) with droppable
     * taking into considerations (+ horizontal).
     */
    $.maxRight = left + horizontal;
  }

  isParenOverflowX() {
    const parentBottom =
      this.activeParent!.offset.top + this.activeParent!.offset.height;

    const elemOverflowX = parentBottom > window.innerHeight;

    return elemOverflowX;
  }

  /**
   * Assigns new ACTIVE_PARENT: parent who contains dragged
   *
   * @param element -
   */
  private assignActiveParent(element: CoreInstanceInterface) {
    /**
     * Assign instance ACTIVE_PARENT which represents droppable. Then
     * assign owner parent so we have from/to.
     */
    this.activeParent = element;

    /**
     * Add flag for undo method so we can check which  parent is being
     * transformed and which is not.
     */
    this.isOutActiveParent = false;

    if (this.opts.scroll.enable) {
      this.opts.scroll.enable = this.opts.scroll.enable
        ? this.isParenOverflowX()
        : false;
    }
  }
}

export default Base;
