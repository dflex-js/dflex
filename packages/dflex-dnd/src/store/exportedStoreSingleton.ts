import DFlexDnDExportedStore from "./DFlexDnDExportedStore";

function createStoreInstance() {
  const store = DFlexDnDExportedStore.getInstance();

  return store;
}

export default createStoreInstance();
