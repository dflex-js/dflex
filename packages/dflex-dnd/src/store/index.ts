// eslint-disable-next-line import/prefer-default-export
export { default as exportedStoreSingleton } from "./exportedStoreSingleton";
export { default as store } from "./storeSingleton";
