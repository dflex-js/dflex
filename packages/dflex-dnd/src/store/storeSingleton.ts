import { canUseDOM } from "@dflex/utils";
import { DFlexDnDStore } from "../LayoutManager";

declare global {
  // eslint-disable-next-line
  var $DFlex: DFlexDnDStore;
}

function createStoreInstance() {
  const store = DFlexDnDStore.getInstance();

  if (__DEV__) {
    if (canUseDOM()) {
      if (!globalThis.$DFlex) {
        globalThis.$DFlex = store;
      }
    }
  }

  return store;
}

export default createStoreInstance();
