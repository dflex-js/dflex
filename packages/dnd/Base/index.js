import Base from "./src";

export default Base;
