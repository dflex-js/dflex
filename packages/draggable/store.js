import CoreInstance from "@dflex/core-instance/src";
import AbstractStore from "@dflex/store/src/AbstractStore";

class DraggableStoreImp extends AbstractStore {
  register(elmInstance) {
    super.register(elmInstance, CoreInstance);
  }
}

let store;

// eslint-disable-next-line func-names
export default (function () {
  if (!store) {
    store = new DraggableStoreImp();
  }

  return store;
})();
