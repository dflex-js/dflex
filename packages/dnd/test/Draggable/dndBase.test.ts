import store from "../../DnDStore";
import Base from "../Base";

const ref = document.createElement("div");

const childInstance1 = {
  id: "id-0",
  depth: 0,
  ref,
};

const childInstance2 = {
  id: "id-1",
  depth: 0,
  ref,
};

const parentInstance = {
  id: "id-p-0",
  depth: 1,
  ref,
};
const DRAGGED_ELM = "draggedElm";

beforeAll(() => {
  store.register(childInstance1);
  store.register(childInstance2);

  store.register(parentInstance);
});

let base;

describe("DND - PKG: Base", () => {
  describe("Testing child", () => {
    beforeAll(() => {
      const START_CLIENT_X = 10;
      const START_CLIENT_Y = 20;

      const elementInstance = store.getElmTreeById(childInstance1.id);

      base = new Base(elementInstance, {
        x: START_CLIENT_X,
        y: START_CLIENT_Y,
      });
    });

    describe("Testing flags and relations", () => {
      it("Checks node siblings", () => {
        expect(base.isSingleton).toBe(false);
      });

      it("Checks node parent", () => {
        expect(base.isOrphan).toBe(false);
      });

      it("Checks parent list", () => {
        expect(base.parentsList).toBe(parentInstance.id);
      });

      it("Checks siblings list", () => {
        expect(base.siblingsList).toStrictEqual([
          childInstance1.id,
          childInstance2.id,
        ]);
      });

      it("Detects activeParent", () => {
        expect(base.activeParent.id).toBe(parentInstance.id);
      });
    });

    describe("Threshold", () => {
      it("Calculates threshold Offset", () => {
        expect(base[DRAGGED_ELM].thresholdOffset).toMatchSnapshot();
      });

      it("Calculates thresholds for dragged as dropped", () => {
        expect(base.thresholds.dragged).toMatchSnapshot();
      });

      it("Calculates thresholds for parent as dropped", () => {
        expect(base.thresholds.parents).toMatchSnapshot();
      });
    });
  });

  describe("Testing parent", () => {
    beforeAll(() => {
      const START_CLIENT_X = 10;
      const START_CLIENT_Y = 20;

      const elementInstance = store.getElmTreeById(parentInstance.id);

      base = new Base(elementInstance, {
        x: START_CLIENT_X,
        y: START_CLIENT_Y,
      });
    });

    describe("Testing flags and relations", () => {
      it("Checks node siblings", () => {
        expect(base.isSingleton).toBe(true);
      });

      it("Checks node parent", () => {
        expect(base.isOrphan).toBe(true);
      });

      it("Checks parent list", () => {
        expect(base.parentsList).toBe(undefined);
      });

      it("Checks siblings list", () => {
        expect(base.siblingsList).toBe(undefined);
      });

      it("Checks siblings list", () => {
        expect(base.isSingleton).toBe(true);
      });

      it("Detects activeParent", () => {
        expect(base.activeParent).toBe(undefined);
      });
    });
  });
});
