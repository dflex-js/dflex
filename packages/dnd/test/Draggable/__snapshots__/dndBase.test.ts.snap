// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DND - PKG: Base Testing child Threshold Calculates threshold Offset 1`] = `
Object {
  "horizontal": 0,
  "vertical": Object {
    "third": 0,
    "twoThirds": 0,
  },
}
`;

exports[`DND - PKG: Base Testing child Threshold Calculates thresholds for dragged as dropped 1`] = `
Object {
  "maxBottom": 0,
  "maxLeft": 0,
  "maxRight": 0,
  "maxTop": 0,
}
`;

exports[`DND - PKG: Base Testing child Threshold Calculates thresholds for parent as dropped 1`] = `
Object {
  "id-p-0": Object {
    "maxBottom": 0,
    "maxLeft": 0,
    "maxRight": 0,
    "maxTop": 0,
  },
}
`;
